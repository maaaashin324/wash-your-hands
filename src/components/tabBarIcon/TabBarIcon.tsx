import React from 'react';
// eslint-disable-next-line
import { MaterialIcons } from '@expo/vector-icons';
import COLORS from '@/constants/colors';

type TabBarIconProps = {
  name: string;
  focused: boolean;
};

const TabBarIcon: React.FC<TabBarIconProps> = ({ name, focused }) => {
  return (
    <MaterialIcons
      name={name}
      size={26}
      color={focused ? COLORS.tabIconSelected : COLORS.tabIconDefault}
    />
  );
};

export default TabBarIcon;
