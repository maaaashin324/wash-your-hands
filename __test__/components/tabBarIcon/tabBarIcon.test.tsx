import React from 'react';
import renderer from 'react-test-renderer';
import TabBarIcon from '@/components/tabBarIcon';

jest.mock('@expo/vector-icons', () => ({
  MaterialIcons: 'MaterialIcons',
}));

describe('TabBarIcon', () => {
  test('should be rendered with main theme color if focused', () => {
    const tree = renderer.create(<TabBarIcon name="home" focused />);
    expect(tree).toMatchSnapshot();
  });

  test('should be rendered with default color if not focused', () => {
    const tree = renderer.create(<TabBarIcon name="home" focused={false} />);
    expect(tree).toMatchSnapshot();
  });
});
