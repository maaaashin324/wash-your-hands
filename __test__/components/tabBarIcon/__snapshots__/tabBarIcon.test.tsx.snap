// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon should be rendered with default color if not focused 1`] = `
<MaterialIcons
  color="#ccc"
  name="home"
  size={26}
/>
`;

exports[`TabBarIcon should be rendered with main theme color if focused 1`] = `
<MaterialIcons
  color="#40CED6"
  name="home"
  size={26}
/>
`;
